[WIP] modified for One-Liner deployment

em3s opened the pull request:
As I think, for #6, the code in this PR is required.

emeth-kim committed:
[WIP] modified for One-Liner deployment

jongwook commented:
I think, it would be better to introduce a method like `runInstall()` to cover both deploy and client mode execution. I will soon make the patch and add here.

em3s committed:
support Kerberos (#5)

Also fixes #3, where Hive configuration was not being picked up

* Set UGI configuration after initializing hadoop conf

* Some modifications for Kerberos.

* cosmetic changes

* added hadoop.security.* for One-Liner Deploy

* add "hadoop.security" prefix to the defaultKeys

* Pass the Hive configuration to `HiveConf` properly

To pass the Hive configuration to `HiveConf` properly,
all the configuration which starts with "hive." are passed by `System.setProperty`.
Then, they are read in `HiveConf.applySystemProperties`.

* copy spark configurations to the system property so that SparkHadoopUtil does not pick up empty authentication settings

* Hive example more readable

jongwook committed:
refactored buildAssembly()
